### PR TITLE
HDDS-3807. Propagate raft log disks info to SCM from datanode.

### PR DESCRIPTION
**Pull request by bshashikant**
## What changes were proposed in this pull request?

Add ratis volume info to  node report

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-3807

## How was this patch tested?

Added unit tests.

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/XceiverServerSpi.java [hunk 1]
```diff
@@ -83,4 +85,12 @@ default void removeGroup(HddsProtos.PipelineID pipelineId)
    * @return list of report for each pipeline.
    */
   List<PipelineReport> getPipelineReport();
+
+  /**
+   * Get storage report for the XceiverServer instance.
+   * @return list of report for each storage location.
+   */
+  default List<MetadataStorageReportProto> getStorageReport() throws IOException {
```
Comment (elek) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/XceiverServerSpi.java:93]: It seems to be a good time the remove XceiverServerSpi interface. Originally we had two separated implementation to connect to the datanode. Today we have only one. One interface is used between the client and the datanode, and the other one between datanode and ratis (datanode). As this example shows, the two interface shouldn't be the same.

**Comment by elek** [comment]
Checkstyle violation seems to be reported.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/XceiverServerSpi.java (hunk 1 above)
Comment (bshashikant) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/XceiverServerSpi.java:93]: Definitely, yes..
Filed https://issues.apache.org/jira/browse/HDDS-3963 for the same

**Comment by elek** [comment]
Merging it as checkstyle issues are fixed. Thanks @bshashikant the contribution.

**Pull request merged**